diff --git a/python/podio/frame.py b/python/podio/frame.py
index f318551..43365ec 100644
--- a/python/podio/frame.py
+++ b/python/podio/frame.py
@@ -40,8 +40,7 @@ def _determine_supported_parameter_types():
     """
     idx, typename = idx_and_type
     if typename == 'float':
-      cpp_type = cppyy.gbl.std.tuple_element[idx, podio.SupportedGenericDataTypes].type
-      if cppyy.typeid(cpp_type).name() == 'd':
+      if idx == 3:
         return 'double'
       return 'float'
     if typename == 'str':
