diff --git a/CMakeLists.txt b/CMakeLists.txt
index e74e40f..e8b24e2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,6 +35,8 @@ PODIO_GENERATE_DATAMODEL(edm4eic edm4eic.yaml headers sources
 PODIO_ADD_DATAMODEL_CORE_LIB(edm4eic "${headers}" "${sources}"
   OUTPUT_FOLDER ${CMAKE_CURRENT_BINARY_DIR}
 )
+# It's our responsibility to link to upstream datamodel
+target_link_libraries(edm4eic PRIVATE EDM4HEP::edm4hep)
 
 PODIO_ADD_ROOT_IO_DICT(edm4eicDict edm4eic "${headers}" src/selection.xml
   OUTPUT_FOLDER ${CMAKE_CURRENT_BINARY_DIR}
